packages/boilerplates/BoilerplatesLoader.ts

```diff
@@ -0,0 +1,184 @@
+
+import { dependencyParsers } from '@dolittle/tooling.common.dependencies';
+import { Folders } from '@dolittle/tooling.common.utilities';
+import * as FsExtra from 'fs-extra';
+import path from 'path';
+import { Logger } from 'winston';
+import { artifactsBoilerplateType, ArtifactsBoilerplate, BaseBoilerplate, Boilerplate, boilerplateContentFolderName, IBoilerplatesLoader, Scripts, BoilerplatesConfig } from './internal';
+
+const binaryFiles = [
+    '.jpg',
+    '.png',
+    '.obj',
+    '.dll',
+    '.bin',
+    '.exe',
+    '.ttf'
+];
+
+/**
+ * Represents the manager of boiler plates
+ */
+export class BoilerplatesLoader implements IBoilerplatesLoader {
+    private _loadedBoilerplates!: BaseBoilerplate[];
+
+    /**
+     * Initializes a new instance of {BoilerplatesManager}
+     * @param {BoilerplatesConfig} boilerplatesConfig
+     * @param {Folders} folders
+     * @param {typeof FsExtra} fileSystem
+     * @param {Logger} logger
+     * @param {typeof Handlebars} handlebars
+     */
+    constructor(private _boilerplatesConfig: BoilerplatesConfig, private _folders: Folders, private _fileSystem: typeof FsExtra, private _logger: Logger) {
+        if (! this._fileSystem.existsSync(this._boilerplatesConfig.path)) {
+            this._boilerplatesConfig.store = this._boilerplatesConfig.store;
+        }
+    }
+    /**
+     * @inheritdoc
+     *
+     * @memberof BoilerplatesLoader
+     */
+    needsReload = true;
+    /**
+     * @inheritdoc
+     *
+     * @readonly
+     * @type {string}
+     * @memberof BoilerplatesLoader
+     */
+    get boilerplatesPath(): string {return this._boilerplatesConfig.path;}
+    /**
+     * @inheritdoc
+     *
+     * @readonly
+     * @type {BaseBoilerplate[]}
+     * @memberof BoilerplatesLoader
+     */
+    get loadedBoilerplates(): BaseBoilerplate[] {
+        if (!this._loadedBoilerplates || this.needsReload) return this.load();
+        return this._loadedBoilerplates;
+    }
+
+    /**
+     * @inheritdoc
+     * 
+     * @memberof BoilerplatesLoader
+     */
+    load(): BaseBoilerplate[] {
+        this._loadedBoilerplates = [];
+        let boilerplatesConfigObject: any = this._boilerplatesConfig.store;
+
+        Object.keys(boilerplatesConfigObject).forEach(key => {
+            let folderPath = path.resolve(boilerplatesConfigObject[key]);
+            if (!this._fileSystem.existsSync(folderPath)) {
+                this._logger.info(`Boilerplate path '${folderPath}' does not exist. Removing entry from boilerplates configuration`);
+                delete boilerplatesConfigObject[key];
+                this._boilerplatesConfig.store = boilerplatesConfigObject;
+            }
+            else this._loadedBoilerplates.push(this.readBoilerplateFromFolder(folderPath));
+        });
+        this.needsReload = false;
+        return this._loadedBoilerplates;
+    }
+    
+    private readBoilerplateFromFolder(folder: string): BaseBoilerplate {
```

Comment:
No need to include Boilerplate in the name; this type is dealing only with Boilerplates. You could even stretch the naming here to just be `readFrom`, especially since it is a private method - the parameter name is folder.